controllers/terminal/controllers/terminal_controller.go

```diff
@@ -48,14 +69,167 @@ type TerminalReconciler struct {
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(terminal, service, r.Scheme); err != nil {
+		logger.Debug("SetControllerReference error: %v", err)
+		return err
+	}
+
+	if err := r.Create(ctx, service); err != nil {
+		logger.Debug("create service error: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func createDeployment(ctx context.Context, r *TerminalReconciler, terminal *terminalv1.Terminal, req ctrl.Request) error {
+	deployment := &appsv1.Deployment{}
+	err := r.Get(ctx, req.NamespacedName, deployment)
+	if err != nil && !errors.IsNotFound(err) {
+		logger.Debug("get deployment error: %v", err)
+		return err
+	}
+
+	if err == nil {
+		// delete previous deployment
+		if err = r.Delete(ctx, deployment); err != nil {
+			logger.Debug("delete deployment error: %v", err)
+			return err
+		}
+	}
+
+	labelsMap := buildLabelsMap(terminal)
+	var (
+		objectMeta      metav1.ObjectMeta
+		selector        *metav1.LabelSelector
+		templateObjMeta metav1.ObjectMeta
+		ports           []corev1.ContainerPort
+		envs            []corev1.EnvVar
+	)
+
+	objectMeta = metav1.ObjectMeta{
+		Name:      terminal.Name,
+		Namespace: terminal.Namespace,
+	}
+	selector = &metav1.LabelSelector{
+		MatchLabels: labelsMap,
+	}
+	templateObjMeta = metav1.ObjectMeta{
+		Labels: labelsMap,
+	}
+	ports = []corev1.ContainerPort{
+		{
+			Name:          "http",
+			Protocol:      corev1.ProtocolTCP,
+			ContainerPort: 8080,
+		},
+	}
+	envs = []corev1.EnvVar{
+		{Name: "APISERVER", Value: r.config.Host},
+		{Name: "USER_TOKEN", Value: terminal.Spec.Token},
+		{Name: "NAMESPACE", Value: terminal.Namespace},
+		{Name: "USER_NAME", Value: terminal.Spec.User},
+	}
+
+	deployment = &appsv1.Deployment{
+		ObjectMeta: objectMeta,
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32(DefaultTTYDeployReplicas),
+			Selector: selector,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: templateObjMeta,
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "tty",
+							Image: DefaultKubeTTYImage,
+							Ports: ports,
+							Env:   envs,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(terminal, deployment, r.Scheme); err != nil {
+		logger.Debug("SetControllerReference error: %v", err)
+		return err
+	}
+
+	if err := r.Create(ctx, deployment); err != nil {
+		logger.Debug("create deployment error: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func buildLabelsMap(terminal *terminalv1.Terminal) map[string]string {
+	labelsMap := map[string]string{
+		"TerminalID":   terminal.Name,
+		"TerminalUser": terminal.Spec.User,
+	}
+	return labelsMap
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *TerminalReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.config = mgr.GetConfig()
+	r.recorder = mgr.GetEventRecorderFor("sealos-terminal-controller")
 	return ctrl.NewControllerManagedBy(mgr).
```

Comment:
Do you want to trigger the owner's modification?  ex Watches(&source.Kind{Type: &appsv1.Deployment{}}, owner)